odo/tests/test_into.py

```diff
@@ -20,6 +20,12 @@ def test_into_append():
     assert result is lst
 
 
+def test_into_append_failure():
+    with pytest.raises(TypeError):
+        nd = np.array([1, 2, 3])
```

Comment:
keep this outside of the `with` statement, as we only want to catch errors triggered by the `odo` call